### PR TITLE
[zh-cn]: Update translation of String.slice()

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

Same as the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#13975

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/string/slice/index.md [hunk 1]
```diff
@@ -5,64 +5,87 @@ slug: Web/JavaScript/Reference/Global_Objects/String/slice
...
-返回一个从原字符串中提取出来的新字符串
+一个包含提取的字符串片段的新字符串。
 
 ## 描述
 
-`slice()` 从一个字符串中提取字符串并返回新字符串。在一个字符串中的改变不会影响另一个字符串。也就是说，`slice` 不会修改原字符串（只会返回一个包含了原字符串中部分字符的新字符串）。
+`slice()` 方法从一个字符串中提取文本并返回一个新的字符串。对它们中的一个的文本进行的更改不会影响另一个字符串。
 
-`slice()` 提取的新字符串包括`beginIndex`但不包括 `endIndex`。下面有两个例子。
+`slice()` 方法提取的文本是从 `indexStart` 开始，但不包括 `indexEnd`。例如，`str.slice(1, 4)` 提取的是第二个字符到第四个字符（字符的索引为 `1`、`2` 和 `3`）。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/slice/index.md:34]: It (the first sentence) is something different from the en-US document.

File: files/zh-cn/web/javascript/reference/global_objects/string/slice/index.md [hunk 2]
```diff
@@ -5,64 +5,87 @@ slug: Web/JavaScript/Reference/Global_Objects/String/slice
...
 
-例 1：`str.slice(1, 4)` 提取第二个字符到第四个字符（被提取字符的索引值（index）依次为 1、2，和 3）。
-
-例 2：`str.slice(2, -1)` 提取第三个字符到倒数第一个字符。
+- 如果 `indexStart >= str.length`，则返回一个空字符串。
+- 如果 `indexStart < 0`，则索引从字符串末尾开始计数。更正式地说，在这种情况下，子字符串从 `max(indexStart + str.length, 0)` 开始。
+- 如果 `indexStart` 被省略、为 undefined，或无法转换为数字（使用 {{jsxref('Number', 'Number(indexStart)')}}），则将其视为 `0`。
+- 如果 `indexEnd` 被省略、为 undefined，或无法转换为数字（使用 {{jsxref('Number', 'Number(indexEnd)')}}），或者 `indexEnd >= str.length`，则 `slice()` 提取到字符串的末尾。
+- 如果 `indexEnd < 0`，则索引从字符串末尾开始计数。更正式地说，在这种情况下，子字符串从 `max(indexEnd + str.length, 0)` 结束。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/slice/index.md:40]:
```suggestion
- 如果 `indexEnd < 0`，则索引从字符串末尾开始计数。更正式地说，在这种情况下，子字符串在 `max(indexEnd + str.length, 0)` 处结束。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/slice/index.md [hunk 3]
```diff
@@ -5,64 +5,87 @@ slug: Web/JavaScript/Reference/Global_Objects/String/slice
...
+- 如果经过负值标准化后（即 `indexEnd` 表示位于 `indexStart` 之前的字符），`indexEnd <= indexStart`，则返回一个空字符串。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/slice/index.md:41]:
```suggestion
- 在标准化负值后，如果 `indexEnd <= indexStart`（即 `indexEnd` 表示位于 `indexStart` 之前的字符），则返回一个空字符串。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/slice/index.md [hunk 4]
```diff
@@ -5,64 +5,87 @@ slug: Web/JavaScript/Reference/Global_Objects/String/slice
...
 
 ## 示例
 
-### 使用 `slice()` 创建一个新的字符串
+### 使用 slice() 创建一个新的字符串
 
 下面例子使用 `slice()` 创建了一个新字符串。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/string/slice/index.md:47]:
```suggestion
以下示例使用 `slice()` 创建了一个新字符串。
```

File: files/zh-cn/web/javascript/reference/global_objects/string/slice/index.md (hunk 1 above)
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/string/slice/index.md:34]:
```suggestion
`slice()` 方法提取直到但不包括 `indexEnd` 的文本。例如，`str.slice(1, 4)` 提取的是第二个字符到第四个字符（字符的索引为 `1`、`2` 和 `3`）。
```

**Review by yin1999 — APPROVED** [review]
Thanks

**Pull request merged**